internal/testrunner/runners/pipeline/test_result.go

```diff
@@ -146,13 +146,27 @@ func unmarshalTestResult(body []byte) (*testResult, error) {
 func marshalTestResultDefinition(result *testResult) ([]byte, error) {
 	var trd testResultDefinition
 	trd.Expected = result.events
-	body, err := json.MarshalIndent(&trd, "", "    ")
+	body, err := normalizeFields(json.Marshal(trd))
 	if err != nil {
 		return nil, errors.Wrap(err, "marshalling test result definition failed")
 	}
 	return body, nil
 }
 
+// normalizeFields ensures that field order remains consistent independent
+// of field order returned by ES to minimize diff noise during changes.
+func normalizeFields(msg []byte, err error) ([]byte, error) {
```

Comment:
It's a bit tangled that `marshalTestResultDefinition` requires to call `json.Marshal` before and then it calls few other `json.*` operations. What do you think about normalizing `testResultDefinition` instead?

Reply:
I had a look at that; making the original unmarshal operation into a `[]interface{}` in `testResultDefinition` which would have that effect, but there is a significant amount of code that depends on the presence of `[]byte` JSON during the tests. I'm not wildly happy with this approach, but it is the least invasive; if it were in a section that was hotter, I think it would likely be worth looking at disentangling, but that's not the case.

Reply:
I see the complexity now. Maybe just put `json.Marshal` inside `normalizeFields` and rename it to `marshalNormalized`?

Reply:
Done